examples/quaternions.dx

```diff
@@ -23,8 +23,8 @@ q
 ' ## Quaternion magnitude
  A quaternion is basically a 4 dimensional vector.
 
-as_vec = \Q{x,y,z,w}. [x, y, z, w]
-from_vec = \[x, y, z, w]. Q{x,y,z,w}
+as_vec = \(Q{x,y,z,w}). [x, y, z, w]
```

Comment:
Do _all_ patterns have to be wrapped in parens?

Reply:
In lambda and def arguments, currently yes.

In this case, we would need to disambiguate between the intent, which is a pattern matching the `Q` constructor with `x, y, z, w` fields, and an alternate interpretation as an explicit argument named `Q` followed by a list of implicit arguments named `x, y, z, w`.  The parens serve to do that.  Though we could certainly consider a more elaborate disambiguation rule.